miqa/core/models/project.py

```diff
@@ -77,6 +77,26 @@ def get_user_role(self, user):
             key=lambda perm: perm_order.index(perm) if perm in perm_order else -1,
         )[-1]
 
+    def get_status(self):
+        tier_2_reviewers = [
+            user.id for user in get_users_with_perms(self, only_with_perms_in=['tier_2_reviewer'])
+        ]
+        complete_scans_per_exp = [
+            len(
+                [
+                    scan
+                    for scan in exp.scans.all()
+                    if scan.decisions.count() > 0
+                    and scan.decisions.latest('created').creator.id in tier_2_reviewers
+                ]
+            )
+            for exp in self.experiments.all()
+        ]
+        return {
+            'total_scans': sum([exp.scans.count() for exp in self.experiments.all()]),
+            'total_complete': sum(complete_scans_per_exp),
+        }
```

Comment:
Does this need to be added to the TS Project type?

Reply:
Good point. Added.

Reply:
The above block looks like it may contain n+1 query issues. Can you check out the SQL queries generated by this function and verify one way or the other?

Reply:
I've admittedly never tried looking directly at SQL queries from django QuerySets. How can I dig down to see that?

Reply:
The django-debug-toolbar is the way to do it for server rendered pages, but since we're using an SPA, that won't work. There are a couple ways to do it that I like to do. If it's straightforward, you can use `manage.py shell_plus --print-sql`, and in the shell, call this function. If for some reason it's not practical to invoke this method in the shell, you can use `manage.py runserver_plus --print-sql` (which involves changing your docker-compose.override.yml if you're using docker-compose).

Reply:
Thanks, I'll try it out!

Reply:
My previous attempt made far too many database hits, and it was slowing down the response time. In dfcec53, I was able to make use of `prefetch_related` to reduce this effect.